File: DotNet/PopulationFitness/PopulationFitness/Models/Genes/Fitness/ReverseSearch.cs
using System;

namespace PopulationFitness.Models.Genes.Fitness
{
    public class ReverseSearch : Search
    {

        public override Search FindNext(PopulationComparison comparison)
        {
            ReverseSearch next = new ReverseSearch();
            next.Increment(Increment()).Max(Max()).Min(Min());
            switch (comparison)
            {
                case PopulationComparison.TooLow:
                    next.Max(Current());
                    break;
                case PopulationComparison.TooHigh:
                    next.Min(Current());
                    break;
                default:
                    return null;
            }
            double current = next.Current();
            if (Math.Abs(next.Max() - current) < Increment()) return null;
            if (Math.Abs(next.Min() - current) < Increment()) return null;
            return next;
        }
    }
}

File: src/main/java/uk/edu/populationfitness/models/genes/fitness/ReverseSearch.java
package uk.edu.populationfitness.models.genes.fitness;

import uk.edu.populationfitness.models.population.PopulationComparison;

public class ReverseSearch extends Search {
    public Search findNext(PopulationComparison comparison){
        ReverseSearch next = new ReverseSearch();
        next.increment(increment()).max(max()).min(min());
        switch(comparison){
            case TooLow:
                next.max(current());
                break;
            case TooHigh:
                next.min(current());
                break;
            default:
                return null;
        }
        double current = next.current();
        if (Math.abs(next.max() - current) < increment()) return null;
        if (Math.abs(next.min() - current) < increment()) return null;
        return next;
    }
}
